lambda/src/main/scala/pricemigrationengine/model/AmendmentData.scala

```diff
@@ -158,11 +158,50 @@ object AmendmentData {
       case Some(_)          => Right(0)
     }
 
-  def combinePrices(pricings: Seq[ZuoraPricing]): BigDecimal =
-    applyDiscountAndThenSum(
-      discountPercentage = pricings.map(_.discountPercentage).find(_ > 0),
-      beforeDiscount = pricings.flatMap(_.price)
-    )
+  /**
+    * Combines prices over the given billing period.
+    *
+    * TODO: this assumes that the prices are per month,
+    * and that billing periods of over a month are priced in the same ratio.
+    * This is only a safe assumption for vouchers.
+    */
+  private def combinePrices(
+      ratePlanChargeAndPricings: Seq[RatePlanChargeAndPricing]
+  ): Either[AmendmentDataFailure, BigDecimal] = {
+
+    def price(ratePlanChargeAndPricing: RatePlanChargeAndPricing): Either[AmendmentDataFailure, Option[BigDecimal]] = {
+      def multiplier(ratePlanCharge: ZuoraRatePlanCharge) = ratePlanCharge.billingPeriod match {
+        case Some("Month")       => Right(1)
+        case Some("Quarter")     => Right(3)
+        case Some("Semi_Annual") => Right(6)
+        case Some("Annual")      => Right(12)
+        case other =>
+          Left(AmendmentDataFailure(s"Rate plan charge '${ratePlanCharge.number}' has unknown billing period: $other"))
+      }
+      ratePlanChargeAndPricing.pricing.price.filter(_ > 0) match {
+        case None => Right(None)
+        case Some(price) =>
+          multiplier(ratePlanChargeAndPricing.ratePlanCharge) map (multiplier => Some(price * multiplier))
```

Comment:
I am probably just missing something but isnt all of the pricing already in terms of the billing period of the subscription?

Ie is this multiplier needed?